src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java

```diff
@@ -89,6 +77,47 @@ public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos,
             Vec3d startPosition = new Vec3d(start).addVector(0.5, 0.5, 0.5);
             Vec3d endPosition = new Vec3d(pos).addVector(0.5, 0.8, 0.5);
             EffectManager.instance.fireSparkEffect(worldIn, startPosition, endPosition);
+            this.spawnParticles(worldIn, pos);
+        }
+    }
+
+    private void spawnParticles(World worldIn, BlockPos pos) {
+        Random random = worldIn.rand;
+        double d0 = 0.0625D;
+
+        for (int i = 0; i < 6; ++i) {
+            double d1 = (double) ((float) pos.getX() + random.nextFloat());
+            double d2 = (double) ((float) pos.getY() + random.nextFloat());
+            double d3 = (double) ((float) pos.getZ() + random.nextFloat());
+
+            if (i == 0 && !worldIn.getBlockState(pos.up()).isOpaqueCube()) {
+                d2 = (double) pos.getY() + 0.0625D + 1.0D;
+            }
+
+            if (i == 1 && !worldIn.getBlockState(pos.down()).isOpaqueCube()) {
+                d2 = (double) pos.getY() - 0.0625D;
+            }
+
+            if (i == 2 && !worldIn.getBlockState(pos.south()).isOpaqueCube()) {
+                d3 = (double) pos.getZ() + 0.0625D + 1.0D;
+            }
+
+            if (i == 3 && !worldIn.getBlockState(pos.north()).isOpaqueCube()) {
+                d3 = (double) pos.getZ() - 0.0625D;
+            }
+
+            if (i == 4 && !worldIn.getBlockState(pos.east()).isOpaqueCube()) {
+                d1 = (double) pos.getX() + 0.0625D + 1.0D;
+            }
+
+            if (i == 5 && !worldIn.getBlockState(pos.west()).isOpaqueCube()) {
```

Comment:
Redstone ore only do this when they are walked on or clicked; should you add a blockstate and do this?